privateCA/create_certs.sh

```diff
@@ -0,0 +1,273 @@
+    --secret-id ${CLIENT_CERT_ARN} \
+    --query SecretString \
+    --output text > ${BACKUP_CERTS_DIR}/apigee_client_cert.pem
+
+aws secretsmanager get-secret-value \
+    --secret-id ${CLIENT_SANDBOX_KEY_ARN} \
+    --query SecretString \
+    --output text > ${BACKUP_KEYS_DIR}/apigee_client_cert_sandbox.key
+aws secretsmanager get-secret-value \
+    --secret-id ${CLIENT_SANDBOX_CERT_ARN} \
+    --query SecretString \
+    --output text > ${BACKUP_CERTS_DIR}/apigee_client_cert_sandbox.pem
+
+echo "Creating new combined truststore files for upload"
+
+aws s3api head-object --bucket ${TRUSTSTORE_BUCKET_NAME} --key ${BUCKET_PREFIX}_truststore.pem || NOT_EXIST=true
+if [ $NOT_EXIST ]; then
+  echo "" > ${BACKUP_CERTS_DIR}/s3_${BUCKET_PREFIX}_truststore.pem
+else
+    aws s3 cp s3://${TRUSTSTORE_BUCKET_NAME}/${BUCKET_PREFIX}_truststore.pem ${BACKUP_CERTS_DIR}/s3_${BUCKET_PREFIX}_truststore.pem 
+fi
+
+aws s3api head-object --bucket ${TRUSTSTORE_BUCKET_NAME} --key ${BUCKET_PREFIX}_sandbox-truststore.pem || NOT_EXIST=true
+if [ $NOT_EXIST ]; then
+  echo "" > ${BACKUP_CERTS_DIR}/s3_${BUCKET_PREFIX}_sandbox_truststore.pem
+else
+    aws s3 cp s3://${TRUSTSTORE_BUCKET_NAME}/${BUCKET_PREFIX}_sandbox-truststore.pem ${BACKUP_CERTS_DIR}/s3_${BUCKET_PREFIX}_sandbox_truststore.pem 
+fi
+
+
+cat ${BACKUP_CERTS_DIR}/s3_${BUCKET_PREFIX}_truststore.pem ${CERTS_DIR}/${CA_NAME}.pem > ${CERTS_DIR}/${BUCKET_PREFIX}_truststore.pem
+cat ${BACKUP_CERTS_DIR}/s3_${BUCKET_PREFIX}_sandbox_truststore.pem ${CERTS_DIR}/${CA_NAME}.pem > ${CERTS_DIR}/${BUCKET_PREFIX}_sandbox_truststore.pem
+
+
+if [ "$DRY_RUN" = "false" ]; then
+    echo "Setting new keys in secrets manager"
+    read -p "Press any key to resume or press ctrl+c to exit ..."
+    aws secretsmanager put-secret-value \
+        --secret-id ${CA_KEY_ARN} \
+        --secret-string file://${KEYS_DIR}/${CA_NAME}.key
+    aws secretsmanager put-secret-value \
+        --secret-id ${CA_CERT_ARN} \
+        --secret-string file://${CERTS_DIR}/${CA_NAME}.pem
+
+    aws secretsmanager put-secret-value \
+        --secret-id ${CLIENT_KEY_ARN} \
+        --secret-string file://${KEYS_DIR}/apigee_client_cert.key
+    aws secretsmanager put-secret-value \
+        --secret-id ${CLIENT_CERT_ARN} \
+        --secret-string file://${CERTS_DIR}/apigee_client_cert.pem
+
+    aws secretsmanager put-secret-value \
+        --secret-id ${CLIENT_SANDBOX_KEY_ARN} \
+        --secret-string file://${KEYS_DIR}/apigee_client_cert_sandbox.key
+    aws secretsmanager put-secret-value \
+        --secret-id ${CLIENT_SANDBOX_CERT_ARN} \
+        --secret-string file://${CERTS_DIR}/apigee_client_cert_sandbox.pem
+
+    echo "Going to create new truststore files on S3"
+    read -p "Press any key to resume or press ctrl+c to exit ..."
+
+    aws s3 cp ${CERTS_DIR}/truststore.pem s3://${TRUSTSTORE_BUCKET_NAME}/truststore.pem
+    aws s3 cp ${CERTS_DIR}/sandbox_truststore.pem s3://${TRUSTSTORE_BUCKET_NAME}/sandbox-truststore.pem
```

Comment:
these two lines need changing as for prescription-tracker it should upload to s3://${TRUSTSTORE_BUCKET_NAME}/clinical-tracker-truststore.pem and clinical-tracker-sandbox-truststore.pem

Reply:
The script has been changed as follows
```
cat ${BACKUP_CERTS_DIR}/s3_${FILE_PREFIX}_truststore.pem ${CERTS_DIR}/${CA_NAME}.pem > ${CERTS_DIR}/${FILE_PREFIX}_truststore.pem
cat ${BACKUP_CERTS_DIR}/s3_${FILE_PREFIX}_sandbox_truststore.pem ${CERTS_DIR}/${CA_NAME}.pem > ${CERTS_DIR}/${FILE_PREFIX}_sandbox_truststore.pem
```
Where
```
$ ./create_certs.sh -e dev -a "clinical tracker"
Going to create mutual TLS certs with these details
AWS_PROFILE: 
CERT_PREFIX: dev-
SECRET_OUTPUT_PREFIX: ClinicalTracker
BUCKET_PREFIX: clinical-tracker
FILE_PREFIX: clinical_tracker
DRY_RUN: true
Press any key to resume or press ctrl+c to exit ...
```